dwave/optimization/include/dwave-optimization/array.hpp

```diff
@@ -723,94 +746,107 @@ class Array {
     const std::string& format() const;
 
-    // The number of dimensions the memory represents as an n-dimensional array.
-    // If 0 the buffer points to a single item representing a scalar
+    /// The number of dimensions the memory represents as an n-dimensional array.
+    /// If 0 the buffer points to a single item representing a scalar.
     virtual ssize_t ndim() const = 0;
 
-    // An array of ndim length indicating the shape of the buffer contents
-    // as an n-dimensional array.
-    // Note that this is in terms of the actual type rather than in terms of
-    // num bytes.
-    // If the shape is state-dependent, the first value in shape will be
-    // DYNAMIC_SIZE
+    /// An array of ndim length indicating the shape of the buffer contents
+    /// as an n-dimensional array.
+    /// Note that this is in terms of the actual type rather than in terms of
+    /// num bytes.
     virtual std::span<const ssize_t> shape(const State& state) const = 0;
+
+    /// An array of ndim length indicating the shape of the buffer contents
+    /// as an n-dimensional array.
+    /// Note that this is in terms of the actual type rather than in terms of
+    /// num bytes.
+    /// If the shape is state-dependent, the first value in shape will be
+    /// Array::DYNAMIC_SIZE
     virtual std::span<const ssize_t> shape() const = 0;
 
-    // An array of length ndim giving the number of bytes to skip to get to a
-    // new element in each dimension.
+    /// An array of length ndim giving the number of bytes to skip to get to a
+    /// new element in each dimension.
     virtual std::span<const ssize_t> strides() const = 0;
 
     // Interface methods ******************************************************
 
+    /// Return an iterator to the beginning of the array.
     const_iterator begin(const State& state) const {
         if (contiguous()) return const_iterator(buff(state));
         return const_iterator(buff(state), ndim(), shape().data(), strides().data());
     }
+
+    /// Return an iterator to the end of the array.
     const_iterator end(const State& state) const { return this->begin(state) + this->size(state); }
 
+    /// Return a container-like view over the array.
     const View view(const State& state) const { return View(this, &state); }
 
-    // The number of doubles in the flattened array.
-    // If the size is dependent on the state, should return DYNAMIC_SIZE.
+    /// The number of doubles in the flattened array.
     virtual ssize_t size() const = 0;
-    // State-dependent size must be a positive integer.
+
+    /// The number of doubles in the flattened array.
+    /// If the size is dependent on the state, returns Array::DYNAMIC_SIZE.
     virtual ssize_t size(const State& state) const = 0;
 
-    // Information about how the size of a node is calculated. By default
-    // a node gets its size from itself.
+    /// Information about how the size of a node is calculated. See SizeInfo.
```

Comment:
```suggestion
    /// Information about how the size of a node is calculated. See ``SizeInfo``.
```

Reply:
This is Doxygen, it finds the link (once we actually document `SizeInfo`) only without the double backticks.